packages/relay/src/lib/eth.ts

```diff
@@ -506,6 +507,10 @@ export class EthImpl implements Eth {
           return predefined.INVALID_PARAMETER(0, `Invalid 'value' field in transaction param. Value must be greater than 0`);
         }
       } else {
+        // execute reverted: with no message comes from a contract deployment
+        if ((e.statusCode != 501) && (e instanceof JsonRpcError) && (e.message === EthImpl.executionReverted)) {
```

Comment:
Similar to previous comments I think it'd be best to return valid revert messages for now or else fallback to what we did before

```suggestion
        if (e instanceof JsonRpcError && e.message !== EthImpl.executionReverted)) {
```

Reply:
None of our `JsonRpcError` objects have a `statusCode` field so `(e.statusCode != 501) && (e instanceof JsonRpcError)` shouldn't be true in the case of `predefined.CONTRACT_REVERT

Reply:
It's the opposite `e.statusCode != 501)` will alays be true since in the `predefined.CONTRACT_REVERT` case  `e.statusCode` will be `undefined`

Reply:
Agreed, but 501 was discussed.  Updated to return defaultGas on a 501.

Reply:
Updated condition with mirror node fix.